proposals/0039-debugbreak.md

```diff
@@ -29,99 +36,406 @@ millions of times without issue, but in one instance produces a bad result.
 Conditional breakpoints can be a powerful tool for shader authors to narrow down
 and identify these complex rare-occurring problems.
 
-This proposal introduces a `DebugBreak()` intrinsic, which can be combined with
-an `assert()` macro implementation to provide support for conditional
-breakpoints in shader code.
+Additionally, shader authors need:
+- A way to terminate execution when an unrecoverable error is detected, matching
+  the behavior of `abort()` in C/C++ for implementing proper `assert()` semantics.
+- A way to conditionally enable expensive debug checks only when a debugger is
+  attached, avoiding runtime overhead in production scenarios.
+
+This proposal introduces three intrinsics that together provide a complete
+debugging toolkit for shader development.
 
 ## Proposed solution
 
-This proposal introduces a new HLSL intrinsic `DebugBreak()`, a new header
-`assert.h` which will define the `assert()` macro in a C-compatible interface.
+This proposal introduces three new HLSL intrinsics and a new header `assert.h`
+which will define the `assert()` macro in a C-compatible interface.
+
+### Intrinsics
+
+```hlsl
+void DebugBreak();        // Trigger a breakpoint if debugger attached
+void Abort();             // Terminate execution abnormally  
```

Comment:
While most HLSL functions are `CamelCase` the ones that have direct C equivalents (like the math functions) we tend to match the C spellings. So I think `abort()` would be preferable.

@tex3d thoughts?